src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/GlobalSearchService.kt

```diff
@@ -0,0 +1,195 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.services
+
+import com.google.gson.Gson
+import com.microsoft.applicationinsights.TelemetryClient
+import org.apache.lucene.search.join.ScoreMode
+import org.elasticsearch.action.search.SearchRequest
+import org.elasticsearch.action.search.SearchResponse
+import org.elasticsearch.index.query.BoolQueryBuilder
+import org.elasticsearch.index.query.QueryBuilders
+import org.elasticsearch.search.builder.SearchSourceBuilder
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.Pageable
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.security.AuthenticationHolder
+import uk.gov.justice.digital.hmpps.prisonersearch.services.exceptions.BadRequestException
+
+@Service
+class GlobalSearchService(
+  private val searchClient: SearchClient,
+  private val indexStatusService: IndexStatusService,
+  private val gson: Gson,
+  private val telemetryClient: TelemetryClient,
+  private val authenticationHolder: AuthenticationHolder
+) {
+  companion object {
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun findByGlobalSearchCriteria(globalSearchCriteria: GlobalSearchCriteria, pageable: Pageable): Page<Prisoner> {
+    validateSearchForm(globalSearchCriteria)
+    if (globalSearchCriteria.offenderNo != null) {
+      queryBy(globalSearchCriteria,pageable) { idMatch(it) } onMatch {
+        customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
+        return PageImpl(it.matches, pageable, it.totalHits)
+      }
+    }
+    if (!(globalSearchCriteria.firstName.isNullOrBlank() && globalSearchCriteria.lastName.isNullOrBlank())) {
+      if (globalSearchCriteria.includeAliases) {
+        queryBy(globalSearchCriteria,pageable) { nameMatchWithAliases(it) } onMatch {
+          customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
+          val ret = PageImpl(it.matches, pageable, it.totalHits)
+          return ret
+        }
+      } else {
+        queryBy(globalSearchCriteria,pageable) { nameMatch(it) } onMatch {
+          customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
```

Comment:
Looks like `        queryBy(globalSearchCriteria,pageable) { nameMatch(it) } onMatch {
          customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
          return PageImpl(it.matches, pageable, it.totalHits)
        }`

is always the same, so you could extract out a method where you passing the `nameMatch(it)` or whatever function